packages/addons/addon-depends/moonlight-embedded/patches/02-surround-fix.patch

```diff
@@ -0,0 +1,46 @@
+diff --git a/src/audio/alsa.c b/src/audio/alsa.c
+index 07f5b88..2cfe7cd 100644
+--- a/src/audio/alsa.c
++++ b/src/audio/alsa.c
+@@ -37,15 +37,13 @@ static int alsa_renderer_init(int audioConfiguration, POPUS_MULTISTREAM_CONFIGUR
+   unsigned char alsaMapping[AUDIO_CONFIGURATION_MAX_CHANNEL_COUNT];
+ 
+   /* The supplied mapping array has order: FL-FR-C-LFE-RL-RR-SL-SR
+-   * ALSA expects the order: FL-FR-RL-RR-C-LFE-SL-SR
++   * CoreELEC expects the order: FL-FR-LFE-C-RL-RR-SL-SR
+    * We need copy the mapping locally and swap the channels around.
+    */
+   memcpy(alsaMapping, opusConfig->mapping, sizeof(alsaMapping));
+   if (opusConfig->channelCount >= 6) {
+-    alsaMapping[2] = opusConfig->mapping[4];
+-    alsaMapping[3] = opusConfig->mapping[5];
+-    alsaMapping[4] = opusConfig->mapping[2];
+-    alsaMapping[5] = opusConfig->mapping[3];
++    alsaMapping[2] = opusConfig->mapping[3];
++    alsaMapping[3] = opusConfig->mapping[2];
+   }
```

Comment:
Moonlight Embedded definitely doesn't handle ALSA multi-channel properly today.

I think the correct approach is to try to use `snd_pcm_set_chmap()` to reorder channels to our intended layout. If that fails (due to lack of hardware/driver support), then call `snd_pcm_get_chmap()` and adjust `alsaMapping` to match the fixed hardware layout as best we can.

If you want take a stab at that, you can open a PR against the upstream repo. Otherwise, I'll probably give it a shot when I find time (though I'll probably need help from someone to test it on the affected hardware).

Reply:
No time for this one... If you manage to make some channel mapping code, I'd consider using a parameter to rearrange the channels with possibly a UI solution in Luna.

This project was my initial idea, but since I can't test it, I gave up.
https://github.com/duncanthrax/scream/blob/master/Receivers/unix/alsa.c